include/eeprom/core/messages.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+namespace eeprom {
+namespace message {
+
+/** Eeprom message */
+struct EepromMessage {
+    eeprom::types::address memory_address{0};
+    eeprom::types::data_length length{0};
+    eeprom::types::EepromData data{};
+
+    auto operator==(const EepromMessage&) const -> bool = default;
+};
+
+using ReadResponseCallback = void (*)(const EepromMessage&, void*);
```

Comment:
I'm not using std::function but a plain old "c" style function pointer. So that issue doesn't factor in.

My initial idea was to use a pointer to an interface like:

```
class ResponseHandler {
    public:
        virtual void handle(const EepromMessage&) = 0;
}
```

The plain old function pointer + param makes it possible to bind to an instance function without using a virtual function.

Reply:
Got it. That makes sense.